src/Tribe/Views/V2/Views/Traits/With_Fast_Forward_Link.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Provides methods for a Views that supports fast-forward links.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Views\Traits
+ */
+
+namespace Tribe\Events\Views\V2\Views\Traits;
+
+use Tribe\Events\Views\V2\View_Interface;
+use Tribe__Date_Utils as Dates;
+
+/**
+ * Class With_Fast_Forward_Link
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Views\Traits
+ */
+trait With_Fast_Forward_Link {
+	/**
+	 * Creates a HTML link and "fast forward" message to append to the "no events found" message.
+	 *
+	 * @since 5.1.0
+	 *
+	 * @param bool  $canonical         Whether to return the canonical (pretty) version of the URL or not.
+	 * @param array $passthru_vars     An optional array of query variables that should pass thru the method untouched
+	 *                                 in key and value.
+	 *
+	 * @return string                  The html link and message.
+	 */
+	public function get_fast_forward_link( $canonical = false, array $passthru_vars = [] ) {
+		if ( ! $this->use_ff_link( $canonical, $passthru_vars ) ) {
+			return '';
+		}
+
+		$date      = $this->context->get( 'event_date', $this->context->get( 'today' ) );
+		$cache_key = __METHOD__ . '_' . md5( wp_json_encode( array_merge( [ $date, $canonical ], $passthru_vars ) ) );
+
+		if ( isset( $this->cached_urls[ $cache_key ] ) ) {
+			//return $this->cached_urls[ $cache_key ];
+		}
```

Comment:
* :warning: This comment is 64% valid code; is this commented out code?
* :no_entry_sign: No space found before comment text; expected "// return $this->cached_urls[ $cache_key ];" but found "//return $this->cached_urls[ $cache_key ];"
* :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: f42b7b30c800ecb856f23388980ce03e</sub>